hazelcast/src/main/java/com/hazelcast/sql/impl/SqlServiceImpl.java

```diff
@@ -175,10 +174,6 @@ public SqlResult execute(
         }
 
         try {
-            if (nodeEngine.getClusterService().getClusterVersion().isLessThan(Versions.V5_0)) {
```

Comment:
Isn't it possible that someone will upgrade from 4.x cluster to 5.1?

Reply:
Seems we only check for version equality really. 
https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/instance/impl/DefaultNodeExtension.java#L500

But there is also a cluster version check here: 
https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterStateManager.java#L268

So I guess a rolling upgrade wouldn't really work on 4x to 5x anyway. So this should be fine probably.

Reply:
To be sure, I tried to set the cluster version to 4.2 in 5.1 code. I got this:
```
com.hazelcast.internal.cluster.impl.VersionMismatchException: Node's codebase version 5.1.0 is incompatible with the requested cluster version 4.2

	at com.hazelcast.internal.cluster.impl.ClusterStateManager.validateNodeCompatibleWith(ClusterStateManager.java:261)
	at com.hazelcast.internal.cluster.impl.ClusterStateManager.lockClusterState(ClusterStateManager.java:209)
	at com.hazelcast.internal.cluster.impl.operations.LockClusterStateOp.run(LockClusterStateOp.java:85)
	at com.hazelcast.spi.impl.operationservice.Operation.call(Operation.java:189)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:276)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:249)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:214)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.run(OperationExecutorImpl.java:411)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.runOrExecute(OperationExecutorImpl.java:438)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:600)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:579)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:540)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:240)
	at com.hazelcast.spi.impl.operationservice.impl.OperationServiceImpl.invokeOnTarget(OperationServiceImpl.java:365)
	at com.hazelcast.internal.cluster.impl.ClusterStateManager.lockClusterStateOnAllMembers(ClusterStateManager.java:454)
	at com.hazelcast.internal.cluster.impl.ClusterStateManager.changeClusterState(ClusterStateManager.java:388)
	at com.hazelcast.internal.cluster.impl.ClusterStateManager.changeClusterState(ClusterStateManager.java:361)
	at com.hazelcast.internal.cluster.impl.ClusterServiceImpl.changeClusterVersion(ClusterServiceImpl.java:863)
	at com.hazelcast.internal.cluster.impl.ClusterServiceImpl.changeClusterVersion(ClusterServiceImpl.java:858)
	at com.hazelcast.cluster.Testt.test(Testt.java:19)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
```
So I think we're safe. It would be a nightmare to support old versions without a limit. Marking this as resolved.